backend/src/index.ts

```diff
@@ -136,6 +138,8 @@ class Server {
       }
       await blocks.$updateBlocks();
       await memPool.$updateMempool();
+      blocks.$generateBlockDatabase();
```

Comment:
We need to wrap this in `if (cluster.isMaster) ` so that only one of the nodejs threads are doing the block indexing.

Same thing when a new block is found. I have not confirmed whether this is an issue or not at the moment

Reply:
We aren't using the cluster feature anymore, so I'm fine with deprecating it for now and re-visiting this later - as you said, the only thing that really needs to be clustered is probably the websocket